Uncaught Invariant Violation: Minified React error #130; visit http://facebook.github.io/react/do...

Hi Lukas,

Thanks for the great help. This is not so crucial but technically with irkernel one is able to run ggplot2 but in hydrogen i got this error.

```
Uncaught Invariant Violation: Minified React error #130; visit http://facebook.github.io/react/docs/error-decoder.html?invariant=130&args[]=undefined&args[]= for the full message or use the non-minified dev environment for full errors and additional helpful warnings.
events.js:160
Hide Stack Trace
Invariant Violation: Minified React error #130; visit http://facebook.github.io/react/docs/error-decoder.html?invariant=130&args[]=undefined&args[]= for the full message or use the non-minified dev environment for full errors and additional helpful warnings.
    at reactProdInvariant (/Users/mea/.atom/packages/Hydrogen/node_modules/react-dom/lib/reactProdInvariant.js:31:15)
    at ReactCompositeComponentWrapper.instantiateReactComponent [as _instantiateReactComponent] (/Users/mea/.atom/packages/Hydrogen/node_modules/react-dom/lib/instantiateReactComponent.js:74:250)
    at ReactCompositeComponentWrapper.performInitialMount (/Users/mea/.atom/packages/Hydrogen/node_modules/react-dom/lib/ReactCompositeComponent.js:367:22)
    at ReactCompositeComponentWrapper.mountComponent (/Users/mea/.atom/packages/Hydrogen/node_modules/react-dom/lib/ReactCompositeComponent.js:258:21)
    at Object.mountComponent (/Users/mea/.atom/packages/Hydrogen/node_modules/react-dom/lib/ReactReconciler.js:46:35)
    at mountComponentIntoNode (/Users/mea/.atom/packages/Hydrogen/node_modules/react-dom/lib/ReactMount.js:104:32)
    at ReactReconcileTransaction.perform (/Users/mea/.atom/packages/Hydrogen/node_modules/react-dom/lib/Transaction.js:140:20)
    at batchedMountComponentIntoNode (/Users/mea/.atom/packages/Hydrogen/node_modules/react-dom/lib/ReactMount.js:126:15)
    at ReactDefaultBatchingStrategyTransaction.perform (/Users/mea/.atom/packages/Hydrogen/node_modules/react-dom/lib/Transaction.js:140:20)
    at Object.batchedUpdates (/Users/mea/.atom/packages/Hydrogen/node_modules/react-dom/lib/ReactDefaultBatchingStrategy.js:62:26)
    at Object.batchedUpdates (/Users/mea/.atom/packages/Hydrogen/node_modules/react-dom/lib/ReactUpdates.js:97:27)
    at Object._renderNewRootComponent (/Users/mea/.atom/packages/Hydrogen/node_modules/react-dom/lib/ReactMount.js:320:18)
    at Object._renderSubtreeIntoContainer (/Users/mea/.atom/packages/Hydrogen/node_modules/react-dom/lib/ReactMount.js:401:32)
    at Object.render (/Users/mea/.atom/packages/Hydrogen/node_modules/react-dom/lib/ReactMount.js:422:23)
    at ResultView.addResult (/Users/mea/.atom/packages/Hydrogen/lib/result-view.js:193:14)
    at /Users/mea/.atom/packages/Hydrogen/lib/main.js:273:12
    at ZMQKernel.onIOMessage (/Users/mea/.atom/packages/Hydrogen/lib/zmq-kernel.js:355:7)
    at Socket.<anonymous> (/Users/mea/.atom/packages/Hydrogen/node_modules/jmp/index.js:347:17)
    at emitMany (events.js:127:13)
    at Socket.emit (events.js:201:7)
    at Socket._emitMessage (/Users/mea/.atom/packages/Hydrogen/node_modules/zeromq/lib/index.js:631:15)
    at Socket._flushRead (/Users/mea/.atom/packages/Hydrogen/node_modules/zeromq/lib/index.js:642:10)
    at Socket._flushReads (/Users/mea/.atom/packages/Hydrogen/node_modules/zeromq/lib/index.js:678:15)
    at _zmq.onReadReady (/Users/mea/.atom/packages/Hydro
```

here it is thanks,

Hydrogen: Could not enable dev tools Error: Cannot find module 'mobx-react-devtools'(…)
utils.js [sm]:75 Hydrogen: findCodeBlock: 16
utils.js [sm]:75 Hydrogen: KernelManager: startKernelFor: r
utils.js [sm]:75 Hydrogen: ZMQKernel: @kernelProcess: ChildProcess {domain: null, _events: EventHandlers, _eventsCount: 0, _maxListeners: undefined, _closesNeeded: 3…}
utils.js [sm]:75 Hydrogen: ZMQKernel: monitor shellSocket
utils.js [sm]:75 Hydrogen: ZMQKernel: monitor controlSocket
utils.js [sm]:75 Hydrogen: ZMQKernel: monitor ioSocket
utils.js [sm]:75 Hydrogen: ZMQKernel: controlSocket connected
utils.js [sm]:75 Hydrogen: ZMQKernel: shellSocket connected
utils.js [sm]:75 Hydrogen: ZMQKernel: ioSocket connected
utils.js [sm]:75 Hydrogen: ZMQKernel: all main sockets connected
utils.js [sm]:75 Hydrogen: KernelManager: startKernelFor: kernel for r connected
utils.js [sm]:75 Hydrogen: clearBubblesOnRow: 16
utils.js [sm]:75 Hydrogen: Kernel.execute: x=14
utils.js [sm]:75 Hydrogen: IO message: Message {idents: Array[1], header: Object, parent_header: Object, metadata: Object, content: Object…}
utils.js [sm]:75 Hydrogen: IO message: Message {idents: Array[1], header: Object, parent_header: Object, metadata: Object, content: Object…}
utils.js [sm]:75 Hydrogen: shell message: Message {idents: Array[0], header: Object, parent_header: Object, metadata: Object, content: Object…}
utils.js [sm]:75 Hydrogen: ResultView: Show status container
utils.js [sm]:75 Hydrogen: IO message: Message {idents: Array[1], header: Object, parent_header: Object, metadata: Object, content: Object…}
utils.js [sm]:75 Hydrogen: findCodeBlock: 17
utils.js [sm]:75 Hydrogen: clearBubblesOnRow: 17
utils.js [sm]:75 Hydrogen: Kernel.execute: x=matrix(0,3,5)
utils.js [sm]:75 Hydrogen: IO message: Message {idents: Array[1], header: Object, parent_header: Object, metadata: Object, content: Object…}
utils.js [sm]:75 Hydrogen: IO message: Message {idents: Array[1], header: Object, parent_header: Object, metadata: Object, content: Object…}
utils.js [sm]:75 Hydrogen: shell message: Message {idents: Array[0], header: Object, parent_header: Object, metadata: Object, content: Object…}
utils.js [sm]:75 Hydrogen: ResultView: Show status container
utils.js [sm]:75 Hydrogen: IO message: Message {idents: Array[1], header: Object, parent_header: Object, metadata: Object, content: Object…}
utils.js [sm]:75 Hydrogen: findCodeBlock: 18
utils.js [sm]:75 Hydrogen: clearBubblesOnRow: 18
utils.js [sm]:75 Hydrogen: Kernel.execute: library(ggplot2)
utils.js [sm]:75 Hydrogen: IO message: Message {idents: Array[1], header: Object, parent_header: Object, metadata: Object, content: Object…}
utils.js [sm]:75 Hydrogen: IO message: Message {idents: Array[1], header: Object, parent_header: Object, metadata: Object, content: Object…}
utils.js [sm]:75 Hydrogen: shell message: Message {idents: Array[0], header: Object, parent_header: Object, metadata: Object, content: Object…}
utils.js [sm]:75 Hydrogen: ResultView: Show status container
utils.js [sm]:75 Hydrogen: IO message: Message {idents: Array[1], header: Object, parent_header: Object, metadata: Object, content: Object…}
utils.js [sm]:75 Hydrogen: findCodeBlock: 19
utils.js [sm]:75 Hydrogen: clearBubblesOnRow: 19
utils.js [sm]:75 Hydrogen: Kernel.execute: qplot(1:3,2:4)
utils.js [sm]:75 Hydrogen: IO message: Message {idents: Array[1], header: Object, parent_header: Object, metadata: Object, content: Object…}
utils.js [sm]:75 Hydrogen: IO message: Message {idents: Array[1], header: Object, parent_header: Object, metadata: Object, content: Object…}
utils.js [sm]:75 Hydrogen: IO message: Message {idents: Array[1], header: Object, parent_header: Object, metadata: Object, content: Object…}
utils.js [sm]:75 Hydrogen: ResultView: Add result Object {data: Object, stream: "pyout"}
utils.js [sm]:75 Hydrogen: ResultView: Hide status container
events.js:160 Uncaught Invariant Violation: Minified React error #130; visit http://facebook.github.io/react/docs/error-decoder.html?invariant=130&args[]=undefined&args[]= for the full message or use the non-minified dev environment for full errors and additional helpful warnings.reactProdInvariant @ reactProdInvariant.js:31instantiateReactComponent @ instantiateReactComponent.js:74performInitialMount @ ReactCompositeComponent.js:367mountComponent @ ReactCompositeComponent.js:258mountComponent @ ReactReconciler.js:46mountComponentIntoNode @ ReactMount.js:104perform @ Transaction.js:140batchedMountComponentIntoNode @ ReactMount.js:126perform @ Transaction.js:140batchedUpdates @ ReactDefaultBatchingStrategy.js:62batchedUpdates @ ReactUpdates.js:97_renderNewRootComponent @ ReactMount.js:320_renderSubtreeIntoContainer @ ReactMount.js:401render @ ReactMount.js:422addResult @ result-view.js [sm]:193(anonymous function) @ main.js [sm]:273onIOMessage @ zmq-kernel.js [sm]:355(anonymous function) @ index.js:347emitMany @ events.js:127emit @ events.js:201Socket._emitMessage @ index.js:631Socket._flushRead @ index.js:642Socket._flushReads @ index.js:678_zmq.onReadReady @ index.js:297

--- a/lib/result-view.js
+++ b/lib/result-view.js
@@ -175,13 +175,17 @@ export default class ResultView {
       container = this.resultContainer;
     }
 
+    const bundle = new ImmutableMap(result.data);
+    const mimeType = richestMimetype(bundle, displayOrder, transforms);
+    if (!mimeType) {
+      log("ResultView: Cannot render", bundle);
+      return;
+    }
+    const Transform = transforms.get(mimeType);
+
     log("ResultView: Hide status container");
     this._hasResult = true;
     this.statusContainer.classList.add("hidden");
-
-    const bundle = new ImmutableMap(result.data);
-    const mimeType = richestMimetype(bundle, displayOrder, transforms);
-    const Transform = transforms.get(mimeType);
 
     // If transforms use unsafe eval, we might need to use loophole due to Atom's CSP:
     // allowUnsafeEval(() => {
@@ -190,9 +194,7 @@ export default class ResultView {
     //   });
     // });
     const div = document.createElement("div");
-    if (mimeType) {
-      ReactDOM.render(<Transform data={bundle.get(mimeType)} />, div);
-    }
+    ReactDOM.render(<Transform data={bundle.get(mimeType)} />, div);
 
     if (mimeType === "text/plain") {
       this.element.classList.remove("rich");

--- a/lib/result-view.js
+++ b/lib/result-view.js
@@ -190,7 +190,9 @@ export default class ResultView {
     //   });
     // });
     const div = document.createElement("div");
-    ReactDOM.render(<Transform data={bundle.get(mimeType)} />, div);
+    if (mimeType) {
+      ReactDOM.render(<Transform data={bundle.get(mimeType)} />, div);
+    }
 
     if (mimeType === "text/plain") {
       this.element.classList.remove("rich");

